pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTTypeParameter.java

```diff
@@ -62,6 +64,13 @@ public ASTType getTypeBoundNode() {
         return getFirstChildOfType(ASTType.class);
     }
 
+    /**
+     * Returns the node to which this type parameter belongs.
+     */
+    public TypeParamOwnerNode getOwner() {
+        return (TypeParamOwnerNode) jjtGetParent().jjtGetParent();
```

Comment:
This is a bit surprising in comparison the formal parameters - in ASTFormalParameter, the owner node is the direct parent and not the grand parent. Maybe something to consider later?

ASTFormalParameter::getOwnerList() vs. ASTTypeParameter::getOwner() ...

Reply:
That's true, I'll add a TODO on the wiki to review that